### PR TITLE
[ALCA] [CLANG] Fixes warnings reported by clang 14

### PR DESCRIPTION
**Pull request by smuzaffar**
clang 14 warning fixes needed for `alca` packages
- removal of set but unused variables
- use of bitwise `&` for boolean operands

### CODE REVIEW

**Comment by tvami** [comment]
@cmsbuild , please test

File: CalibMuon/DTCalibration/plugins/DTTTrigCorrectionFirst.cc [hunk 1]
```diff
@@ -86,7 +85,6 @@ void DTTTrigCorrectionFirst::endJob() {
     tTrigMap->get((*sl)->id(), ttrigMean, ttrigSigma, kfactor, DTTimeUnits::ns);
     if (ttrigMean < ttrigMax && ttrigMean > ttrigMin) {
       average2 += (ttrigMean - average) * (ttrigMean - average);
-      average2Sigma += (ttrigSigma - averageSigma) * (ttrigSigma - averageSigma);
```
Comment (tvami) [inline CalibMuon/DTCalibration/plugins/DTTTrigCorrectionFirst.cc:89]: @namapane is this supposed to be used for something?

File: Calibration/HcalCalibAlgos/src/hcalCalib.cc [hunk 2]
```diff
@@ -503,10 +503,8 @@ void hcalCalib::GetCoefFromMtrxInvOfAve() {
 
     std::map<Int_t, Float_t>::iterator n_it = (aveHitE[iEta]).begin();
 
-    Float_t sumRawE = 0;
     for (; n_it != (aveHitE[iEta]).end(); ++n_it) {
       (n_it->second) *= norm;
-      sumRawE += (n_it->second);
```
Comment (tvami) [inline Calibration/HcalCalibAlgos/src/hcalCalib.cc:509]: @bsunanda is this supposed to be used for something?

File: CalibMuon/DTCalibration/plugins/DTTTrigCorrectionFirst.cc (hunk 1 above)
Comment (smuzaffar) [reply CalibMuon/DTCalibration/plugins/DTTTrigCorrectionFirst.cc:89]: @tvami , this was used in a `cout` statement but https://github.com/cms-sw/cmssw/pull/36534 has removed its usage

Comment (namapane) [reply CalibMuon/DTCalibration/plugins/DTTTrigCorrectionFirst.cc:89]: indeed, should be safe to remove.

**Comment by tvami** [comment]
+alca
 * tests pass, subsystem experts are notified of the changes

**Comment by perrotta** [comment]
+1

**Pull request merged**